### PR TITLE
Remove support for no inst prefix

### PR DESCRIPTION
**Pull request by jkonecny12**
Anaconda kernel boot arguments without `inst.` prefix won't be accepted and used. Only warning will be printed that these were removed and you should use the `inst.` variant.

This will happen in Dracut and also in Anaconda.

This is implementation of [Fedora system wide change](https://fedoraproject.org/wiki/Changes/Ignore_Anaconda_kernel_boot_parameters_without_inst_prefix).

### CODE REVIEW

File: anaconda.py [hunk 1]
```diff
@@ -304,9 +304,8 @@ def setup_environment():
         log.info("Using updates in /tmp/updates/ from %s", opts.updateSrc)
 
     # warn users that they should use inst. prefix all the time
-    for arg in depr:
-        stdout_log.warning("Boot argument '%s' is deprecated and will be removed in the future. "
-                           "Please use 'inst.%s' instead.",
+    for arg in removed_no_inst_args:
+        stdout_log.warning("Boot argument '%s' was removed. Please use 'inst.%s' instead.",
```
Comment (poncovka) [inline anaconda.py:308]: I don't think we can still claim these boot options as ours.

File: dracut/parse-anaconda-options.sh [hunk 2]
```diff
@@ -58,6 +58,12 @@ check_removed_arg() {
     fi
 }
 
+check_removed_no_inst_arg() {
+    local removed_arg="$1" new_arg="$2"
+    check_removed_arg "$removed_arg" "All usage of Anaconda boot arguments without 'inst.' prefix \
+was removed. Please use $new_arg instead."
```
Comment (poncovka) [inline dracut/parse-anaconda-options.sh:64]: Same here.

**Review by poncovka — COMMENTED** [review]
Since we are dropping the support for the boot options without the `inst.` prefix, we can just remove everything that is related to that. I don't think we should continue to show warnings, because these options might belong to someone else now.

**Comment by jkonecny12** [comment]
I agree but I would like to leave it there for at least some time and make another PR with complete removal in the future. It's one small thing we can do for our consumers.

**Review by poncovka — APPROVED** [review]
Looks good to me. Thanks! Please, open an issue for the removal of the warnings about the removal :-P

File: anaconda.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def setup_environment():
     # do this early so we can set flags before initializing logging
     from pyanaconda.flags import flags
     from pyanaconda.core.kernel import kernel_arguments
-    (opts, depr) = parse_arguments(boot_cmdline=kernel_arguments)
+    (opts, removed_no_inst_args) = parse_arguments(boot_cmdline=kernel_arguments)
```
Comment (VladimirSlavik) [inline anaconda.py:265]: Is this correct? There will be no more deprecated arguments?

**Comment by jstodola** [comment]
Hi, this change will likely break installations of new composes in Beaker. Could you please wait before merging this PR?
Bug https://bugzilla.redhat.com/show_bug.cgi?id=1907566 tracks the Beaker change.
Thank you :-)

File: dracut/parse-anaconda-options.sh [hunk 4]
```diff
@@ -77,35 +83,35 @@ check_removed_arg askmethod "Use an appropriate 'inst.repo=' argument instead."
...
 
 # lang & keymap
-warn_renamed_arg "lang" "inst.lang"
-warn_renamed_arg "keymap" "inst.keymap"
+check_removed_no_inst_arg "lang" "inst.lang"
+check_removed_no_inst_arg "keymap" "inst.keymap"
 
 # debug
-warn_renamed_arg "debug" "inst.debug"
+check_removed_no_inst_arg "debug" "inst.debug"
```
[the remark this answers is not in the document]
Comment (jkonecny12) [reply dracut/parse-anaconda-options.sh:90]: Good point. Thanks for input!

File: anaconda.py (hunk 3 above)
Comment (jkonecny12) [reply anaconda.py:265]: It wasn't used for anything else than just `inst.` prefix so I don't see a reason to keep it universal anymore. My next step will be to remove this completely.

**Comment by jkonecny12** [comment]
I'll give them a bit more time but as I wrote to bugzilla, we can't wait too long. It has to be merged before Tue 2021-02-09 because of change completion deadline.

File: dracut/parse-anaconda-options.sh (hunk 4 above)
[the remark this answers is not in the document]
Comment (jkonecny12) [reply dracut/parse-anaconda-options.sh]: Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Remove `debug` warning noted by @mopsfelder . Thanks!
* Add bug to commits.

**Review by VladimirSlavik — APPROVED** [review]

**Comment by jkonecny12** [comment]
> Looks good to me. Thanks! Please, open an issue for the removal of the warnings about the removal :-P

Issue opened https://issues.redhat.com/browse/INSTALLER-2363 .

**Comment by jkonecny12** [comment]
**UPDATED:**
* Update warning in stage2 to be more generic. The warning is now suggest to user instead of telling him that he's using the parameter wrong. Point is that this argument may not be ours (like in case of `debug`).

**Comment by jkonecny12** [comment]
@VladimirSlavik could you please look on the new message in commit c16d17d if it makes sense to you? Or do you have a suggestion for improvement?

**Comment by VladimirSlavik** [comment]
The commit message looks to me perfectly understandable.

**Review by poncovka — APPROVED** [review]
It looks great! Thanks!

File: anaconda.py [hunk 5]
```diff
@@ -304,10 +304,13 @@ def setup_environment():
         log.info("Using updates in /tmp/updates/ from %s", opts.updateSrc)
 
     # warn users that they should use inst. prefix all the time
-    for arg in depr:
-        stdout_log.warning("Boot argument '%s' is deprecated and will be removed in the future. "
-                           "Please use 'inst.%s' instead.",
+    for arg in removed_no_inst_args:
+        stdout_log.warning("The '%s' kernel boot argument detected. "
+                           "Did you want to use 'inst.%s' instead?",
```
Comment (VladimirSlavik) [inline anaconda.py:309]: Sorry, now I see what you meant. How about:

Kernel boot argument '%s' detected. Did you want to use 'inst.%s' for the installer instead?

Comment (jkonecny12) [reply anaconda.py]: Fixed.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Update message as suggested by @VladimirSlavik . Thanks!

**Review by VladimirSlavik — APPROVED** [review]

**Review by poncovka — APPROVED** [review]

**Comment by jkonecny12** [comment]
These coverage test failures are not because of this change.

**Pull request merged**